src/basic/sysctl-util.c

```diff
@@ -44,7 +44,7 @@ char *sysctl_normalize(char *s) {
         return s;
 }
 
-int sysctl_write(const char *property, const char *value) {
+int sysctl_write(const char *property, const char *value, Hashmap *cache) {
```

Comment:
it think sysctl_write() should keep its pre-existing prototype, but become a static inline around sysctl_write_full() which adds the third param. Then, code in networkd that wants the shadow stuff would use sysctl_write_full() but the rest of the codebase can use sysctl-write as before